crypto-ffi/src/lib.rs

```diff
@@ -236,6 +236,21 @@ impl CoreCrypto {
             .map_err(CryptoError::from)
     }
 
+    pub fn new_external_add_proposal(
+        &self,
+        conversation_id: ConversationId,
+        epoch: u64,
+        key_package: Vec<u8>,
+    ) -> CryptoResult<Vec<u8>> {
+        let kp = KeyPackage::try_from(&key_package[..]).map_err(MlsError::from)?;
+        use core_crypto::prelude::tls_codec::Serialize as _;
+        self.0
+            .new_external_add_proposal(conversation_id, GroupEpoch::from(epoch), kp)?
+            .tls_serialize_detached()
```

Comment:
Check if you can use `to_bytes()` to avoid pulling the tls_codec trait

```suggestion
        self.0
            .new_external_add_proposal(conversation_id, GroupEpoch::from(epoch), kp)?
            .to_bytes()
```